File: inutile/xx.cs
using System;
using System.IO;
using System.Text.Json;
using System.Text.Json.Serialization;

class Program
{
    static void Main()
    {
        var random = new Random();
        var data = new object[10];

        for (int i = 0; i < 10; i++)
        {
            data[i] = new { id = i, value = random.Next(100) };
        }

        var options = new JsonSerializerOptions { WriteIndented = true };
        var json = JsonSerializer.Serialize(data, options);

        File.WriteAllText("data.json", json);
    }
}

File: inutile/xx.kt
import java.io.File
import kotlin.random.Random
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
data class DataItem(val id: Int, val value: Int)

fun main() {
    val data = List(10) { DataItem(it, Random.nextInt(100)) }
    val json = Json { prettyPrint = true }.encodeToString(data)
    File("data.json").writeText(json)
}